### DISCUSSION TITLE
Alternative support for C++ package manager - which one?

### DISCUSSION THREAD

**Discussion by assignUser** [facebookincubator/velox#4992] [Polls]
I think it would be beneficial (and not that much work) to add support for a package manager like conda, conan or nix in addition to the resolve_dependency system. 

This would allow us to install compiled binaries locally and in ci to reduce compile time while still keeping our dependencies separate from system dependencies. In addition it should be easier to add (complex) new dependencies needed for new features (e.g. grpc for gcs support) to keep development momentum up. Compared to adding such dependencies to resolve_dependency can be quite time consuming and hinder exploration of new features.

Additionally  this could allow us to simplify the multiple `setup-*` scripts to avoid having to keep changes aligned across multiple files.

I would like to hear which pms the community favors! I think conda would be a good choice as it is wide spread and a lot of developers already use it. It would also make it easier to later on develop the conda-forge recipe for velox

I did not list vcpkg as that does build from source and thus does not have the advantages in reducing build times (at least without caching).

**Comment by assignUser** [comment]
This would also solve issues like #2758

**Comment by majetideepak** [comment]
How do Conda, Conan, and Nix differ? Can you please provide a short summary?

**Reply by assignUser** [reply]
Both conda and nix are multi-language package managers where as conan is C/C++ only.

Both conda and nix also create isolated build environments, conan only provides the needed packages via a cmake toolchain file without any additional isolation or changes to the environment.

nix has a large focus on reproducibility with a strong 'functional' philosophy underpinning it's design. nix has some very interesting features but is also less  widespread and a bit harder to initially get used to.

**Reply by kgpai** [reply]
In light of conda being very popular in python world, I think it makes sense to go with conda , because of larger support and use base. conan seems more c++ specific, and nix would probably be more harder to get started with. Should our needs change we can re-evaluate these options imo, but conda seems the simplest option for now.
